src/IQFeed.CSharpApiClient/Streaming/Derivative/DerivativeClientFactory.cs

```diff
@@ -17,6 +18,18 @@ public class DerivativeClientFactory
                 derivativeMessageHandler);
         }
 
+        public static DerivativeClient<T> CreateNew<T>(
+            IPAddress host,
+            int port,
+            int bufferSize,
+            IDerivativeMessageHandler<T> derivativeMessageHandler)
+        {
+            return new DerivativeClient<T>(
```

Comment:
Before returning new, add guards to test `host, port and buffersize` for within range, above 0 and etc.

Reply:
I removed these methods and added logic into the `SocketClient` constructor to handle IP addresses. I think that's a cleaner approach.